src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -481,6 +481,9 @@ bool Creature::UpdateEntry(uint32 entry, CreatureData const* data /*= nullptr*/)
         ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_ATTACK_ME, true);
     }
 
+    if (cInfo->flags_extra & CREATURE_FLAG_EXTRA_ROOT)
+        AddUnitState(UNIT_STATE_ROOT);
```

Comment:
AddUnitState is a wrong method to call here, it should be SetControlled(true, UNIT_STATE_ROOT);